src/sqlfluff/core/parser/segments/base.py

```diff
@@ -78,6 +78,21 @@ def __hash__(self):
         return hash((self.edit, self.source_slice.start, self.source_slice.stop))
 
 
+@dataclass()
```

Comment:
Should we consider this? (Mainly for documentation; I assume these objects won't be changed after creation.)
```suggestion
@dataclass(frozen=True)
```

Reply:
Good idea